site/content/docs/contribute/scan-rules.md

```diff
@@ -0,0 +1,105 @@
+
+To see the results of ZAP running against well known vulnerable apps go to [ZAP Scans](/docs/scans/) - any issues that ZAP fails to find on these apps could be a good candidate for fixing.
+
+However it must be stressed that the key focus for ZAP is to find vulnerabilities in real world apps. Scoring better against test apps should be as a consequence of improving the rules for real world apps rather than the end goal.
+
+### Finding the Relevant Scan Rule
+
+You can see the list of current ZAP rules on the [Alert Details](/docs/alerts/) page.
+
+Note that there may be more than one rule that could potentially identify any given vulnerability.
+
+If you are looking to fix a false positive then you should be able to identify the relevant rule from the title.
+
+Once you find the relevant rule select it and you will see a link to the code at the bottom of the page.
+
+If you cannot see one that applies then you might need to create a new scan rule, but ask on the [ZAP Developer Group](https://groups.google.com/group/zaproxy-develop) just to make sure.
+
+### Deciding if Improvements are Possible
+
+You will need to be able to reproduce any vulnerability manually before you can enhance ZAP to find it.
+
+When you have done so you need to decide if it is reasonable for an automated scanner like ZAP to be able to find it.
+
+Many vulnerabilities will require more knowledge or understanding of an app than an automated scanner will be able to deduce.
+If in doubt ask on the [ZAP Developer Group](https://groups.google.com/group/zaproxy-develop).
+
+In a similar way some false positives may still be reasonable for ZAP to report if the only way to tell that they are false positives is by a knowledgeable person.
+
+### Implementation Details
+
+The following blog posts will help you get started with scan rules - they are a bit old but they are mostly still accurate:
+* [Hacking ZAP #3 - Passive scan rules](/blog/2014-04-03-hacking-zap-3-passive-scan-rules/)
+* [Hacking ZAP #4 - Active scan rules](/blog/2014-04-30-hacking-zap-4-active-scan-rules/)
+
+All changes should be covered by unit tests - see [Verifying Your Changes](/docs/developer/verifying-your-changes/).
+
+For scan rules this means that you will need simulate a web server - most of the existing scan rule unit test already do this, so look at the existing tests to see how this can be done.
+
+#### Number of Requests
+
+Active scan rules should limit the number of requests they make based on the Strength set by the user:
+
+* LOW:      Limit to around 6 requests per scan call
+* MEDIUM:   Limit to around 12 requests
+* HIGH:     Limit to around 24 requests
+* INSANE:   No limit, although 1000s wouldn’t be a good idea
+
+### Alert Field Details
+
+The following advice and guidance should be followed for any new alerts raised.
```

Comment:
A link to https://www.zaproxy.org/docs/desktop/ui/dialogs/addalert/ can be added here. It contains some info on other fields like Risk, Confidence and Alert Tags which are not mentioned here.

Reply:
Done :)

Reply:
Thanks!